contracts/Strategy.sol

```diff
@@ -148,106 +138,45 @@ contract Strategy is BaseStrategy {
         isWantIncentivised = _isWantIncentivised;
         isInvestmentTokenIncentivised = _isInvestmentTokenIncentivised;
         leaveDebtBehind = _leaveDebtBehind;
-        require(maxLoss <= 10_000);
         maxLoss = _maxLoss;
     }
 
-    event Cloned(address indexed clone);
-
-    function cloneAaveLenderBorrower(
-        address _vault,
-        address _strategist,
-        address _rewards,
-        address _keeper,
-        address _yVault,
-        bool _isWantIncentivised,
-        bool _isInvestmentTokenIncentivised,
-        string memory _strategyName
-    ) external returns (address newStrategy) {
-        require(isOriginal);
-        // Copied from https://github.com/optionality/clone-factory/blob/master/contracts/CloneFactory.sol
-        bytes20 addressBytes = bytes20(address(this));
-        assembly {
-            // EIP-1167 bytecode
-            let clone_code := mload(0x40)
-            mstore(
-                clone_code,
-                0x3d602d80600a3d3981f3363d3d373d3d3d363d73000000000000000000000000
-            )
-            mstore(add(clone_code, 0x14), addressBytes)
-            mstore(
-                add(clone_code, 0x28),
-                0x5af43d82803e903d91602b57fd5bf30000000000000000000000000000000000
-            )
-            newStrategy := create(0, clone_code, 0x37)
-        }
-
-        Strategy(newStrategy).initialize(
-            _vault,
-            _strategist,
-            _rewards,
-            _keeper,
-            _yVault,
-            _isWantIncentivised,
-            _isInvestmentTokenIncentivised,
-            _strategyName
-        );
-
-        emit Cloned(newStrategy);
+    // Where to route token swaps
+    // Access control is stricter in this method as it will be sent funds
+    function setSwapRouter(ISwap _router) external onlyGovernance {
+        router = _router;
     }
 
-    function _initializeThis(
-        address _yVault,
-        bool _isWantIncentivised,
-        bool _isInvestmentTokenIncentivised,
-        string memory _strategyName
-    ) internal {
-        minReportDelay = 24 * 3600;
-        maxReportDelay = 10 * 24 * 3600;
-        profitFactor = 100;
-        // debtThreshold = 0; It's 0 by default.
-
+    function _initializeThis(address _yVault, string memory _strategyName)
+        internal
+    {
         yVault = IVault(_yVault);
```

Comment:
Add another require to check for initialization just for extra safety

Reply:
Fixed in 756b5bfb75ff747c6347b54bb2eb9c0d44f2756a